### DISCUSSION TITLE
What happens to starred gists or repos when they are deleted?

### DISCUSSION THREAD

**Discussion by markuszeller** [community/community#21615] [New to GitHub]
<p><strong>Topic:</strong></p>
<p>What happens to starred gists or repos when they are deleted?</p>
<p><strong>Situation:</strong></p>
<p>I forked a repo. The original got removed. As the documention says, another fork will be selected as the new “master” and all other forks will get linked to it.</p>
<p><strong>Conclusion:</strong></p>
<p>If there is no other fork, my would become the new master.</p>
<p><strong>Question 1:</strong></p>
<p>But, what happens, when neither me or no one else forked it, but only starred?</p>
<p>Is it silently removed and gone forever?</p>
<p><strong>Question 2:</strong></p>
<p>Is it then better to have a fork instead of star?</p>

**Comment by logankilpatrick** [comment, marked as answer]
<p>Hey <a class="mention" href="/markuszeller">@markuszeller</a>, if a repo is starred and then subsequently deleted, the star will disappear along with the repo once it’s deleted. If you want to preserve a copy of the code, forking seems like a logical alternative. </p>

**Comment by ivankovnatsky** [comment]
<p>should it also remove stars from the user who starred the repo?</p>
<p>because i’m seeing 4 stars that i gave on my personal page, but it only shows 2 starred repos:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/1/3/131e9ef4c6b0ae926908651e13e9704e250b71b1.png" data-download-href="/uploads/short-url/2J8FYf5NgWZ08pw0Ytqmws79c3v.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/75213/181065846-c803869d-baf4-47f9-8d54-c3f44c385653.png" alt="image" data-base62-sha1="2J8FYf5NgWZ08pw0Ytqmws79c3v" width="690" height="432" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/1/3/131e9ef4c6b0ae926908651e13e9704e250b71b1_2_690x432.png, https://aws1.discourse-cdn.com/github/optimized/3X/1/3/131e9ef4c6b0ae926908651e13e9704e250b71b1_2_1035x648.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/1/3/131e9ef4c6b0ae926908651e13e9704e250b71b1_2_1380x864.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/1/3/131e9ef4c6b0ae926908651e13e9704e250b71b1_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">2160×1354 131 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>I was able to get starred repos using api:</p>
<pre><code class="lang-auto">    "full_name": "kanashimia/nixos",
    "full_name": "cole-mickens/nixcfg",
    "full_name": "TechBookHunter/Free-Career-Guide-Books",
    "full_name": "TechBookHunter/Free-DevOps-Books",
</code></pre>
<p>the latter i guess was just removed.</p>
<p>should the stars also be removed from the user profile as well?</p>

**Comment by JRSKelvin** [comment]
Hi @ivankovnatsky may I know which API to list all starred repos?

**Reply by ivankovnatsky** [reply]
https://github.com/orgs/community/discussions/24756.
